cfy_manager/components/manager/manager.py

```diff
@@ -37,86 +36,81 @@
-    logger.notice('Removing Cloudify Manager resources...')
-    remove_files([
-        join(_get_exec_tempdir(), 'cloudify-ctx'),
-    ])
-    logger.notice('Cloudify Manager resources successfully removed')
+class ManagerComponent(BaseComponent):
+    def __init__(self, skip_installation):
+        super(ManagerComponent, self).__init__(skip_installation)
+
+    def _get_exec_tempdir(self):
+        return os.environ.get(constants.CFY_EXEC_TEMPDIR_ENVVAR) or gettempdir()
+
+    def _create_cloudify_user(self):
+        create_service_user(
+            user=constants.CLOUDIFY_USER,
+            group=constants.CLOUDIFY_GROUP,
+            home=constants.CLOUDIFY_HOME_DIR
+        )
+        common.mkdir(constants.CLOUDIFY_HOME_DIR)
+
+    def _create_sudoers_file_and_disable_sudo_requiretty(self):
+        common.remove(constants.CLOUDIFY_SUDOERS_FILE, ignore_failure=True)
+        touch(constants.CLOUDIFY_SUDOERS_FILE)
+        common.chmod('440', constants.CLOUDIFY_SUDOERS_FILE)
+        entry = 'Defaults:{user} !requiretty'.format(user=constants.CLOUDIFY_USER)
+        description = 'Disable sudo requiretty for {0}'.format(
+            constants.CLOUDIFY_USER
+        )
+        add_entry_to_sudoers(entry, description)
+
+    def _get_selinux_state(self):
+        try:
+            return subprocess.check_output(['/usr/sbin/getenforce']).rstrip('\n\r')
+        except OSError as e:
+            logger.warning('SELinux is not installed ({0})'.format(e))
+            return None
+
+    def _set_selinux_permissive(self):
+        """This sets SELinux to permissive mode both for the current session
+        and systemwide.
+        """
+        selinux_state = self._get_selinux_state()
+        logger.debug('Checking whether SELinux in enforced...')
+        if selinux_state == 'Enforcing':
+            logger.info('SELinux is enforcing, setting permissive state...')
+            common.sudo(['setenforce', 'permissive'])
+            replace_in_file(
+                'SELINUX=enforcing',
+                'SELINUX=permissive',
+                '/etc/selinux/config')
+        else:
+            logger.debug('SELinux is not enforced.')
+
+    def _create_manager_resources_dirs(self):
+        resources_root = constants.MANAGER_RESOURCES_HOME
+        common.mkdir(resources_root)
+        common.mkdir(join(resources_root, 'cloudify_agent'))
+        common.mkdir(join(resources_root, 'packages', 'scripts'))
+        common.mkdir(join(resources_root, 'packages', 'templates'))
+
+    def _configure(self):
+        self._create_cloudify_user()
+        self._create_sudoers_file_and_disable_sudo_requiretty()
+        self._set_selinux_permissive()
+        setup_logrotate()
+        self._create_manager_resources_dirs()
+
+    def install(self):
+        logger.notice('Installing Cloudify Manager resources...')
+        self._configure()
+        logger.notice('Cloudify Manager resources successfully installed')
+
+    def configure(self):
```

Comment:
can be moved to the base component

Reply:
Will do after merge as it's not related to this feature specifically